### PR TITLE
main/nginx: upgrade to 1.14.0

### PR DESCRIPTION
**Pull request by andypost**
Ref http://nginx.org/en/CHANGES-1.14

### CODE REVIEW

File: main/nginx/APKBUILD [hunk 1]
```diff
@@ -8,8 +8,8 @@
 #     - CVE-2017-7529
 #
 pkgname=nginx
-pkgver=1.12.2
-pkgrel=6
+pkgver=1.14.0
+pkgrel=0
 # Revision of nginx-tests to use for check().
 _tests_hgrev=cdd44ff602db
```
Comment (TBK) [inline main/nginx/APKBUILD]: If you change it to the latest commit d6daf03478ad, only 2 tests will fail.

Comment (TBK) [reply main/nginx/APKBUILD]: There is also the question of if the new njs module should be added as a subpackage - http://nginx.org/en/docs/njs_about.html

Comment (andypost) [reply main/nginx/APKBUILD]: I think njs needs separate PR

**Comment by andypost** [comment]
> (Wstat: 512 Tests: 3 Failed: 2)

```
./proxy_bind_transparent_capability.t ...... 1/3 
#   Failed test 'transparent'
#   at ./proxy_bind_transparent_capability.t line 72.
#                   undef
#     doesn't match '(?^:X-IP: 127.0.0.2)'
#   Failed test 'no alerts'
#   at lib/Test/Nginx.pm line 82.
#          got: '2018/04/23 22:05:59 [alert] 16431#16431: worker process 16432 exited with fatal code 2 and cannot be respawned'
#     expected: ''
# Looks like you failed 2 tests of 3.
./proxy_bind_transparent_capability.t ...... Dubious, test returned 2 (wstat 512, 0x200)
Failed 2/3 subtests 
```

**Comment by andypost** [comment]
> ./grpc_ssl.t ............................... skipped: OpenSSL too old
> ./geo_ipv6.t ............................... skipped: no inet6 support
> ./image_filter.t ........................... skipped: GD not installed
> ./js.t ..................................... skipped: no njs available

looks makes sense to add `njs`

**Comment by andypost** [comment]
Pushed commit, with njs module I got tests pass locally

**Pull request closed without merging**